bundles/org.openhab.automation.jsscripting/src/main/resources/node_modules/@jsscripting-globals.js

```diff
@@ -0,0 +1,172 @@
+    if (arguments.length === 1) return f;
+
+    var i = 1;
+    var args = arguments;
+    var len = args.length;
+    var str = String(f).replace(formatRegExp, function (x) {
+        if (x === '%%') return '%';
+        if (i >= len) return x;
+        switch (x) {
+            case '%s': return String(args[i++]);
+            case '%d': return Number(args[i++]);
+            case '%j':
+                try {
+                    return stringify(args[i++]);
+                } catch (_) {
+                    return '[Circular]';
+                }
+            // falls through
+            default:
+                return x;
+        }
+    });
+    for (var x = args[i]; i < len; x = args[++i]) {
+        if (x === null || (typeof x !== 'object' && typeof x !== 'symbol')) {
+            str += ' ' + x;
+        } else {
+            str += ' ' + stringify(x);
+        }
+    }
+    return str;
+}
+
+const counters = {};
+const timers = {};
+
+const console = {
+    'assert': function (expression, message) {
+        if (!expression) {
+            log.error(message);
+        }
+    },
+
+    count: function (label) {
+        let counter;
+
+        if (label) {
+            if (counters.hasOwnProperty(label)) {
+                counter = counters[label];
+            } else {
+                counter = 0;
+            }
+
+            // update
+            counters[label] = ++counter;
+            log.debug(format.apply(null, [label + ':', counter]));
+        }
+    },
+
+    debug: function () {
+        log.debug(format.apply(null, arguments));
+    },
+
+    info: function () {
+        log.info(format.apply(null, arguments));
+    },
+
+    log: function () {
+        log.info(format.apply(null, arguments));
+    },
+
+    warn: function () {
+        log.warn(format.apply(null, arguments));
+    },
+
+    error: function () {
+        log.error(format.apply(null, arguments));
+    },
+
+    trace: function (e) {
+        if (Java.isJavaObject(e)) {
+            log.trace(e.getLocalizedMessage(), e);
+        } else {
+            if (e.stack) {
+                log.trace(e.stack);
+            } else {
+                if (e.message) {
+                    log.trace(format.apply(null, [(e.name || 'Error') + ':', e.message]));
+                } else {
+                    log.trace((e.name || 'Error'));
+                }
+            }
+        }
+    },
+
+    time: function (label) {
+        if (label) {
+            timers[label] = System.currentTimeMillis();
+        }
+    },
+    timeEnd: function (label) {
+        if (label) {
+            const now = System.currentTimeMillis();
+            if (timers.hasOwnProperty(label)) {
+                log.info(format.apply(null, [label + ':', (now - timers[label]) + 'ms']));
+                delete timers[label];
+            } else {
+                log.info(format.apply(null, [label + ':', '<no timer>']));
+            }
+        }
+    }
+};
+
+function setTimeout(cb, delay) {
```

Comment:
Ideally we probably should also have a polyfill for `setInterval`!?

Reply:
Agreed.

Reply:
Just added this in, also i changed both types of timers to support arguments to better confirm with standards.